Script/sample_training_script_multi_seq_lstm_classifier.py

```diff
@@ -0,0 +1,211 @@
+with open(registry_path, 'r') as f:
+	w2v_registry = json.load(f)
+
+def initiate_logger(log_path):
+	"""
+	Initialize a logger with file handler and stream handler
+	"""
+	logger = logging.getLogger(__name__)
+	logger.setLevel(logging.INFO)
+	formatter = logging.Formatter('%(asctime)s %(levelname)-s: %(message)s', datefmt='%H:%M:%S')
+	fh = logging.FileHandler(log_path)
+	fh.setLevel(logging.INFO)
+	fh.setFormatter(formatter)
+	logger.addHandler(fh)
+	sh = logging.StreamHandler(sys.stdout)
+	sh.setLevel(logging.INFO)
+	sh.setFormatter(formatter)
+	logger.addHandler(sh)
+	logger.info('===================================')
+	logger.info('Begin executing at {}'.format(time.ctime()))
+	logger.info('===================================')
+	return logger
+
+def train(model, train_inp_tuple, validation_inp_tuple, checkpoint_dir, checkpoint_prefix, device, epoches=5, batch_size=1024, logger=None, epoch_start=0):
+	"""
+	: model (torch.nn.module): model to be trained
+	: train_inp_tuple (list[tuple(str, list[str], list[str])]): list of input for train_data_loader
+		: str: path to label data
+		: list[str]: list of embedding variables
+		: list[str]: list of paths to a pkl file 
+	: validation_inp_tuple (list[tuple(str, list[str], list[str])]): list of input for train_data_loader
+		: str: path to label data
+		: list[str]: list of embedding variables
+		: list[str]: list of paths to a pkl file
+	: checkpoint_dir (str): path to checkpoint directory
+	: checkpoint_prefix (str): prefix of checkpoint file
+	: device (str): device to train the model
+	: epoches (int): number of epoches to train
+	: batch_size (int): size of mini batch
+	: epoch_start (int): if = 0 then train a new model, else load an existing model and continue to train, default 0
+	"""
+	global w2v_registry, model_path
+	gc.enable()
+
+	# Check checkpoint directory
+	if not os.path.isdir(checkpoint_dir): os.mkdir(checkpoint_dir)
+
+	# Load model if not train from scratch
+	if epoch_start != 0:
+		model_artifact_path = os.path.join(checkpoint_dir, '{}_{}.pth'.format(checkpoint_prefix, epoch_start))
+		model.load_state_dict(torch.load(model_artifact_path))
+		if logger: logger.info('Start retraining from epoch {}'.format(epoch_start))
+
+	# Set up loss function and optimizer
+	model.to(device)
+	loss_fn = nn.CrossEntropyLoss()
+	optimizer = torch.optim.Adam(model.parameters())
+
+	# Main Loop
+	for epoch in range(1+epoch_start, epoches+1+epoch_start):
+		if logger:
+			logger.info('=========================')
+			logger.info('Processing Epoch {}/{}'.format(epoch, epoches+epoch_start))
+			logger.info('=========================')
+
+		# Train model
+		model.train()
+		train_running_loss, train_n_batch = 0, 0
+
+		for index, (label_artifact_path, seq_inp_target, seq_inp_path) in enumerate(train_inp_tuple, start=1):
+			train_loader = train_data_loader(label_artifact_path, seq_inp_target, seq_inp_path, w2v_registry)
+			train_iterator = iter(train_loader)
+			while True:
```

Comment:
Why `while True`?

Reply:
Cause it's using an iterator which will throw StopIteration exception when reaching the end